column-line-timeline/columnLineTimeline.js

```diff
@@ -0,0 +1,156 @@
+import * as d3 from 'd3';
+import gChartcolour from 'g-chartcolour';
+
+export function drawBars() {
+    let yScaleL = d3.scaleLinear();
+    let yScaleR = d3.scaleLinear();
+    let xScale0 = d3.scaleTime();
+    let xScale1 = d3.scaleBand();
+    let barWidth = 10;
+    let seriesNamesL = [];
+    let rem = 16;
+    let barColour = d3.scaleOrdinal()
+        .domain(seriesNamesL);
+
+    function bars(parent) {
+
+        parent.attr('transform', d => `translate(${xScale0(d.date)},0)`)
+            .attr('width', barWidth);
+
+        parent.selectAll('rect')
+            .data(d => d.groups)
+            .enter()
+            .append('rect')
+            .attr('class', 'columns')
+            .attr('x', d => xScale1(d.name))
+            .attr('width', () => xScale1.bandwidth())
+            .attr('y', d => yScaleL(Math.max(0, d.value)))
+            .attr('height', d => Math.abs(yScaleL(d.value) - yScaleL(0)))
+            .attr('fill', function(d,i)  { return barColour.range()[i+1]});
+
+    }
+
+    bars.yScaleL = (d) => {
+        if (!d) return yScaleL;
+        yScaleL = d;
+        return bars;
+    };
+    bars.yScaleR = (d) => {
+        if (!d) return yScaleR;
+        yScaleR = d;
+        return bars;
+    };
+    bars.xScale0 = (d) => {
+        if (!d) return xScale0;
+        xScale0 = d;
+        return xScale0;
+    };
+    bars.xScale1 = (d) => {
+        if (!d) return xScale0;
+        xScale0 = d;
+        return xScale0;
+    };
+    bars.seriesNamesL = (d) => {
+        seriesNamesL = d;
+        return bars;
+    };
+    bars.xScale0 = (d) => {
+        if (!d) return xScale0;
+        xScale0 = d;
+        return bars;
+    };
+    bars.xScale1 = (d) => {
+        if (!d) return xScale1;
+        xScale1 = d;
+        return bars;
+    };
+    bars.rem = (d) => {
+        if (!d) return rem;
+        rem = d;
+        return bars;
+    };
+    bars.barWidth = (d) => {
+        if (!d) return barWidth;
+        barWidth = d;
+        return bars;
+    };
+    bars.colourPalette = (d) => {
```

Comment:
Add getter here.

Reply:
That all seems fine so I'll merge that now